libraries/snapshotter/snapshotter.cpp

```diff
@@ -0,0 +1,43 @@
+#include <eosio/chain/controller.hpp>
+#include <eosio/chain/fork_database.hpp>
+#include <eosio/chain/deep_mind.hpp>
+
+using namespace eosio::chain;
+
+//extern "C" void doit(int db_fd, int forkdb_fd) {
+extern "C" int makesnap(char* db_path, char* forkdb_path, char* protocol_features_path, char* snapshot_output_path) {
```

Comment:
Originally instead of paths I wanted to pass in file descriptors because the intent was to sandbox the execution and passing in file descriptors means that the sandbox can disallow opening files. Allowing chainbase & forkdb to accept already open file handles will require some further refactoring of those. But the real problem is the protocol_feature_set which needs to be loaded from a directory and is required for one aspect of snapshot creation. I think it might be worthwhile to reflect on whether what we're doing with these protocol feature files on disk is even needed/appropriate.

(it'd be possible for me to potentially work around it by passing a serialized protocol_feature_set to the function)

Since paths are being passed here, can undo some of the other changes made for chainbase & forkdb which pass a path instead of a directory.